python-threatexchange/threatexchange/cli/hash_cmd.py

```diff
@@ -99,10 +112,31 @@ def execute(self, settings: CLISettings) -> None:
                     f"{self.signal_type.get_name()} "
                     f"does not apply to {self.content_type.get_name()}"
                 )
+
             hashers = [self.signal_type]  # type: ignore  # can't detect intersection types
 
+        if not self.rotations:
+            for file in self.files:
+                for hasher in hashers:
+                    hash_str = hasher.hash_from_file(file)
+                    if hash_str:
+                        print(hasher.get_name(), hash_str)
+            return
+
+        if not issubclass(self.content_type, PhotoContent):
+            raise CommandError(
+                "--rotations flag is only available for Photo content type", 2
+            )
+
         for file in self.files:
-            for hasher in hashers:
-                hash_str = hasher.hash_from_file(file)
-                if hash_str:
-                    print(hasher.get_name(), hash_str)
+            with open(file, "rb") as f:
+                image_bytes = f.read()
+                rotated_images = PhotoContent.all_simple_rotations(image_bytes)
+                for _, rotated_bytes in rotated_images.items():
+                    with tempfile.NamedTemporaryFile() as temp_file:  # Create a temporary file to hold the byte data
```

Comment:
ignorable: What are your thoughts on passing in the bytes via `hash_bytes`? Think about it before reading the spoiler.

<details>
  <summary>Spoiler warning</summary>
  `hash_bytes` will work in most cases, but some file formats are only knowable by their extensions. No file name = no extension hint, and pillow will fail to parse the file.
</details>

Reply:
Then it would be a BytesHasher!